buildenv/jenkins/common/variables-functions

```diff
@@ -0,0 +1,243 @@
+    } else {
+        OPENJ9_BRANCH = ''
+    }
+    
+    OPENJ9_SHA = (params.OPENJ9_SHA != null) ? params.OPENJ9_SHA : ''
+    
+    if ((params.OMR_REPO != null) && (params.OMR_REPO != '')) {
+         OMR_REPO = params.OMR_REPO
+    } else if (VARIABLES.omr_repo != null) {
+         OMR_REPO = VARIABLES.omr_repo
+    } else {
+        OMR_REPO = ''
+    }
+    
+    if ((params.OMR_BRANCH != null) && (params.OMR_BRANCH != '')) {
+        OMR_BRANCH = params.OMR_BRANCH
+    } else if (VARIABLES.omr_branch != null) {
+         OMR_BRANCH = VARIABLES.omr_branch
+    } else {
+        OMR_BRANCH = ''
+    }
+    
+    OMR_SHA = (params.OMR_SHA != null) ? params.OMR_SHA : ''
+    
+    echo "Using OPENJDK_REPO = ${OPENJDK_REPO} OPENJDK_BRANCH = ${OPENJDK_BRANCH} OPENJDK_SHA = ${OPENJDK_SHA}"
+    echo "Using OPENJ9_REPO = ${OPENJ9_REPO} OPENJ9_BRANCH = ${OPENJ9_BRANCH} OPENJ9_SHA = ${OPENJ9_SHA}"
+    echo "Using OMR_REPO = ${OMR_REPO} OMR_BRANCH = ${OMR_BRANCH} OMR_SHA = ${OMR_SHA}"
+}
+
+/*
+* Returns the value of the element identified by the given key or an empty string 
+* if the map contains no mapping for the key. 
+*/
+def get_value(MAP, KEY) {
+    for (item in MAP) {
+        if ("${item.key}" == "${KEY}") {
+            return "${item.value}"
+        }
+    }
+
+    return ''
+}
+
+/*
+* Returns Jenkins credentials ID required to connect to GitHub using 
+* the ssh protocol.
+* Returns empty string if no values is provided in the variables file (not 
+* required for public repositories).
+*/
+def get_git_user_credentials_id () {
+    if (VARIABLES.user != null) {
+        return VARIABLES.user.credentials_id
+    }
+    
+    return ''
+}
+
+/*
+* Sets the NODE variable.
+* Returns the labels identifying nodes by platform suitable to run a Jenkins job.
+*/
+def set_node(job_type) { 
+    // fetch labels (space separated string) for given platform/spec
+    labels = get_value(VARIABLES."${SPEC}".node_labels, job_type)
+    NODE = labels.replaceAll(" ", "&&")
+}
+
+/*
+* Sets variables for a job that builds the OpenJ9 extensions for OpenJDK for a 
+* given platform and version.
+*/
+def set_build_variables() {
+    set_repos_variables()
+    
+    // fetch values per spec and Java version from the variables file
+    BOOT_JDK = get_value(VARIABLES."${SPEC}".boot_jdk, SDK_VERSION)
+    FREEMARKER = VARIABLES."${SPEC}".freemarker
+    RELEASE = get_value(VARIABLES."${SPEC}".release, SDK_VERSION)
+    JDK_IMAGE_DIRS = VARIABLES.jdk_image_dir
+    JDK_FOLDER = get_value(VARIABLES.jdk_image_dir, SDK_VERSION)
+    
+    try{
+        SDK_PREFIX = "OpenJ9-JDK${SDK_VERSION}-PR${ghprbPullId}-${SPEC}-"
+    } catch (e) {
+        SDK_PREFIX = "OpenJ9-JDK${SDK_VERSION}-${SPEC}-" 
+    }
+    
+    SDK_SUFFIX = ".tar.gz"
+    TEST_PREFIX = "test-"
+    TEST_SUFFIX = ".tar.gz"
+}
+
+/*
+* Sets variables for a test job on a given platform.
+*/
+def set_test_variables() {
+    RELEASE = VARIABLES."${SPEC}".release
```

Comment:
I think it is this line, should be the same as line 172. I wonder if we should not be repeating this code. Perhaps move it to a function?

Reply:
@AdamBrousseau ready for review